azure/scope/managedcontrolplane.go

```diff
@@ -527,8 +533,13 @@ func (s *ManagedControlPlaneScope) AgentPoolSpec() azure.AgentPoolSpec {
 		replicas = *s.MachinePool.Spec.Replicas
 	}
 
+	poolName := s.InfraMachinePool.GetName()
+	if s.InfraMachinePool.Spec.Name != nil {
+		poolName = *s.InfraMachinePool.Spec.Name
```

Comment:
we already default Spec.Name to the MachinePool name if it is not set: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/api/v1beta1/azuremanagedmachinepool_webhook.go#L49 so I don't think this is needed.

To avoid nil pointer errors, how about using
`Name:          to.String(s.InfraMachinePool.Spec.Name),`
below?

Reply:
I followed the comment on the `spec.Name` which says its optional and in case its not specified it should use the CR name. 
```
// Name - name of the agent pool. If not specified, CAPZ uses the name of the CR as the agent pool name.
// +optional
```

which probably will never happen due to webhook `failurePolicy: Fail` 
I will revert the change as for now